### DISCUSSION TITLE
what is difference between Concepts and Componants of semantic kernel

### DISCUSSION THREAD

**Discussion by rakshahulle** [microsoft/semantic-kernel#9600] [Q&A] [answered]
I was going through the GitHub repo of Semantic kernel there I found python/samples/concepts folder. It had many concepts like auto function invocation, filters, planner, plugin. So now I am confused between the concept and components of the semantic kernel. Can you please help me to understand this?

**Comment by moonbox3** [comment, marked as answer]
Hello @rakshahulle, concepts/components can be used interchangeably. The `python/samples/concepts` are demonstrating various concepts about how to apply different SK components -- like auto function calling, filters, etc.

The main takeaway is: if you can understand that the Kernel is the central component to Semantic Kernel, that's a great start. The kernel controls the state of an application -- one can add plugins, filters, invoke those functions, etc, via the kernel object. An input to the kernel is an AI service (how you interact with the LLM) and one can define multiple AI services like `AzureChatCompletion` alongside an `AnthropicChatCompletion` service. Each service has its own prompt execution settings (which settings are sent to the model).

To help round things out, I'm including a screenshot from the learn site:

![image](https://github.com/user-attachments/assets/2085dd0f-8ff6-4d70-8c1f-96e66a3aed71)

You can follow the top purple/blue line from left to right, and then the bottom blue/purple line from right to left. This is a high-level flow using SK.

**Comment by rakshahulle** [comment]
Thank you so much this was helpful.
